Code/GraphMol/Descriptors/EEM.cpp

```diff
@@ -0,0 +1,237 @@
+};
+
+
+// Those Parameters change to adapted to the molecule dataset using the optimization method:
+// The best parameters published in the NEEMP article (https://jcheminf.springeropen.com/articles/10.1186/s13321-016-0171-1),
+// are CCD_gen_DE_RMSD_B3LYP_6311G_NPA.par from Additional file 6 was built using the larger dataset with B3LYP_6311G ab initio => 17769 molecules
+// H 1, C 6, N 7, O 8, F 9, P 15, S 16, Cl 17, Br 35
+// Caution: Iodine is not in this training set of molecules!!!
+const float kappa = 0.5125;
+                   //0   1      2   3   4   5   6       7      8      9     10  11  12  13  14  15      16    17    18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35
+const float A1[] = {0.0,2.5473,0.0,0.0,0.0,0.0,2.7221,2.9750,3.1503,2.9976,0.0,0.0,0.0,0.0,0.0,0.0,   2.6511,2.7026,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,2.6263};
+const float A2[] = {0.0,0.0,   0.0,0.0,0.0,0.0,2.7667,2.8895,3.0486,0.0,   0.0,0.0,0.0,0.0,0.0,2.2933,2.6471,   0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0};
+const float A3[] = {0.0,0.0,   0.0,0.0,0.0,0.0,2.6944,3.0240,0.0,   0.0,   0.0,0.0,0.0,0.0,0.0,0.0,   0.0,      0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0};
+                   //0   1      2   3   4   5   6       7      8      9     10  11  12  13  14  15      16    17     18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35
+const float B1[] = {0.0,1.1641,0.0,0.0,0.0,0.0,0.6403,0.9083,1.0577,0.9983,0.0,0.0,0.0,0.0,0.0,0.0,   0.4897,1.1537,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,1.1105};
+const float B2[] = {0.0,0.0,   0.0,0.0,0.0,0.0,0.6513,0.6647,0.8410,0.0,   0.0,0.0,0.0,0.0,0.0,0.5759,0.4512,   0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0};
+const float B3[] = {0.0,0.0,   0.0,0.0,0.0,0.0,0.6776,1.4240,0.0,   0.0,   0.0,0.0,0.0,0.0,0.0,0.0,   0.0,      0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0};
+
+/* simple Parameters trained using the small 500 molecules dataset.
+const float kappa = 0.1960;
+
+const float A1[] = {0.0,2.3594,0.0,0.0,0.0,0.0,2.4541,2.5908,2.7130,0.0,0.0,0.0,0.0,0.0,0.0,0.0,2.3833};
+const float A2[] = {0.0,0.0,0.0,0.0,0.0,0.0,2.4726,2.5409,2.6766,0.0,0.0,0.0,0.0,0.0,0.0,0.0,2.4956};
+const float A3[] = {0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0};
+
+const float B1[] = {0.0,0.5962,0.0,0.0,0.0,0.0,0.2591,0.3316,0.5028,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.4564};
+const float B2[] = {0.0,0.0,0.0,0.0,0.0,0.0,0.2268,0.2319,0.4992,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.1493};
+const float B3[] = {0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0};
+*/
+
+// function to retreive the atomtype value based on the highest bond type of an atom
+// in the publication they don't have access to "Aromatic type"
+double getAtomtype(const RDKit::ROMol mol, const RDKit::Atom *atom) {
```

Comment:
argument should be a mol reference (i.e. `const ROMol &mol`), otherwise you copy the molecule when you call this, which is *slow*. You also don't need to use the `RDKit` namespace here.